knimin/tests/test_ag_edit_barcode.py

```diff
@@ -89,6 +93,105 @@ def test_post(self):
         self.assertEqual(response.code, 200)
         self.assertIn("Error Updating Barcode Info", response.body)
 
+    def test_source_reassignment(self):
+        # Elaine Wolfe found this bug, May 13th:
+        # Manually changing the survey a sample is assigned to is not saved
+        # even after getting the "barcode has successfully been updated"
+        # message. All other field changes (time, sample type, etc.) are
+        # saving properly.
+        self.mock_login_admin()
+
+        # this is a barcode that belongs to a set of ag_kits which has multiple
+        # participant_names, i.e. sources
+        # SELECT barcode
+        # FROM barcodes.barcode
+        # JOIN ag.ag_kit_barcodes USING (barcode)
+        # JOIN ag.ag_kit USING (ag_kit_id)
+        # WHERE ag_login_id IN
+        #       (SELECT ag_login_id
+        #               FROM (SELECT array_agg(participant_name) as sources,
+        #                            count(participant_name) as numsources,
+        #                            ag_login_id FROM ag.ag_login_surveys
+        #                     GROUP BY ag_login_id) as foo
+        #               WHERE numsources > 2);
+        barcode = '000023125'  # current survey_id = "a4f1061f5bac9ae3"
+        details = db.getAGBarcodeDetails(barcode)
+        payload = {'barcode': barcode,
+                   'ag_kit_id': details['ag_kit_id'],
+                   'site_sampled': details['site_sampled'],
+                   'sample_date': details['sample_date'],
+                   'sample_time': details['sample_time'],
+                   'participant_name': details['participant_name'],
+                   'notes': details['notes'],
+                   'environment_sampled': details['environment_sampled'],
+                   'refunded': details['refunded'] or 'N',
+                   'withdrawn': details['withdrawn'] or 'N'}
+        response = self.post('/ag_edit_barcode/', payload)
+        self.assertEqual(response.code, 200)
+        # check that no actual change has happened
+
+        dbinfo = db.getAGBarcodeDetails(barcode)
+        for field in set(payload.keys()) - set(self.none_fields):
```

Comment:
You can create a copy of payload in which you change this values to None, so you actually test that those values are None as you expect, rather than not testing them.